docs: general formatting, fix links, cleanups, rewrites

chuck-sys opened the pull request:
## Details

- Add links to types in internal reference
- Change the unicode directional single-/double-quotes to ascii ones so
that mac users aren't selectively favoured
- Add even more documentation on less documented functions
- Make factory documentation visible when it wasn't for so long
- Rewrite segments of architecture because it no longer reflected the
current state of the project
- Reformat some names to make internal reference links possible

View newly built docs here: https://rocket2.readthedocs.io/en/cheukyin699-update-docs/

chuck-sys committed:
docs: fix formatting, intrinsic links

- Add links to types in internal reference
- Change the unicode directional single-/double-quotes to ascii ones so
that mac users aren't selectively favoured
- Add even more documentation on less documented functions
- Make factory documentation visible when it wasn't for so long
- Rewrite segments of architecture because it no longer reflected the
current state of the project
- Reformat some names to make internal reference links possible

Probably should be in separate commits. But they are all under the
umbrella of general cleanup.

chuck-sys committed:
docs: rm members that should be really private

chuck-sys committed:
docs: add config and attributes

chuck-sys committed:
docs: make docstring consistant

bobheadxi commented:
> Change the unicode directional single-/double-quotes to ascii ones so
that mac users aren't selectively favoured

❤️

bobheadxi commented on app/controller/command/commands/team.py:
```diff
@@ -635,7 +639,8 @@ def refresh_helper(self, user_id) -> ResponseTuple:
         In the event that our local team database is outdated compared to
-        the teams on GitHub, this command can be called to fix things.
+        the teams on GitHub, this command can be called to fix these
+        inconsistencies.
```
This likely ties into https://github.com/ubclaunchpad/rocket2/issues/490 in that we can't remove this command, as we'll always need something to forcibly bring everything in line

cheukyin699 replied:
Probably need to add a `Deprecation Warning` whenever someone calls the command.

bobheadxi replied:
Oh my point was I _dont_ think we can deprecate it, ever, since we need some way to force a state sync (for example, to add new features - essentially a "migration")

bobheadxi commented on conf.py:
```diff
@@ -23,9 +23,9 @@
+version = '2.0'
 # The full version, including alpha/beta/rc tags
-release = '1.0.0'
+release = '2.0.0'
```
do we even do releases?

cheukyin699 replied:
Good question. Actually, there is a slim chance that this would come back and bite us. Technically we do releases. We are currently on v2.1.0. I'm thinking that'd be one of the things we'd talk about in the meeting?

Gonna change this to better reflect, but right now docs are generated (and not saved) on master push, soooo rolling release.

bobheadxi replied:
I dropped an item in the meeting notes - feel free to add stuff! https://docs.google.com/document/d/1KcH2l6nJKyHoCDe6v6Sl0cLWQWKQQmSLqk_kXvZUsRw/edit

bobheadxi commented on config/__init__.py:
```diff
@@ -118,7 +118,7 @@ def __init__(self, missing_config_fields):
         Initialize a new MissingConfigError.
 
-        :param: missing_config_fields List of missing config variables
+        :param missing_config_fields: the missing config variables
```
is there a linter that can catch these documentation errors?

cheukyin699 replied:
I doubt it.

bobheadxi commented on docs/Architecture.rst:
```diff
@@ -3,18 +3,22 @@ Architecture
 
-**What is the ``model`` module?** The ``model`` module is where models
-are constructed. Currently we have ``Team`` and ``User`` models.
+We treat Github itself as the sole source of truth. Any modifications done on the
```
```suggestion
We treat GitHub itself as the sole source of truth. Any modifications done on the
```

etc - though we can also have a separate PR to make all these fixes in one go

Reference: official branding

![image](https://user-images.githubusercontent.com/23356519/95804617-cbe2d380-0d35-11eb-8938-4b836a2ce14d.png)

bobheadxi commented:
This might be because, to center-align the "Rocket 2" header in both GitHub README and the website, I use HTML, which isn't recognized by Sphinx as the top-level header for the page - is there a way to override this? (should be "Rocket 2", not "Usage")

![image](https://user-images.githubusercontent.com/23356519/95804709-106e6f00-0d36-11eb-8f2c-73816e47fe14.png)

Commit:
docs: GitHub

Co-authored-by: Robert Lin <robertlin1@gmail.com>

chuck-sys committed:
docs: minor fixes to readme

- set header of readme correctly so that it is picked up by sphinx
- use links to make README slightly more readable

chuck-sys committed:
Merge branch 'cheukyin699/update-docs' of github.com:ubclaunchpad/rocket2 into cheukyin699/update-docs

cheukyin699 commented:
> Is there a way to override this?

I just noticed. Good question. After a bit of read, I think it's safe to say that, yes, there is a way to override this. And **no**, it's not pretty. See 62a6e82.

bobheadxi commented:
@cheukyin699 sadly, your approach in 62a6e82f517e84fd87880f085db10c36b0980485 doesn't work on GitHub :( that's why I made the entire top section raw HTML previously - so that it would work on GitHub as well (IMO this is a requirement)

![image](https://user-images.githubusercontent.com/23356519/95809768-2a15b380-0d42-11eb-92b4-248c2fb2a724.png)

cheukyin699 commented:
@rob ahhh. Ohhhh. Welp, back to the drawing board. Might not be possible then.

bobheadxi commented:
Let's leave that for another PR - can you revert 62a6e82 and I'll approve?

cheukyin699 commented:
Another option:

```rst
.. raw:: html

	<style>
		h1:nth-of-type(2) {
			display: none;
		}
	<style>
```

But that would be way too hacky for me.

chuck-sys committed:
docs: revert broken gh docs

chuck-sys committed:
docs: revert image that now works

chuck-sys committed:
docs: fix indenting

cheukyin699 commented:
@bobheadxi reverted

bobheadxi reviewed (approved):
thank you!!!!

bobheadxi commented:
Created https://github.com/ubclaunchpad/rocket2/issues/555 as follow-up
